addons/stock_account/models/stock_move_line.py

```diff
@@ -41,6 +41,8 @@ def write(self, vals):
                 if move_line.state != 'done':
                     continue
                 move = move_line.move_id
+                if float_compare(vals['qty_done'], move_line.qty_done, precision_rounding=move.product_uom.rounding) == 0:
+                    continue
                 rounding = move.product_id.uom_id.rounding
                 diff = move.product_uom._compute_quantity(vals['qty_done'] - move_line.qty_done, move.product_id.uom_id)
```

Comment:
Let's make it so we don't have to worry about the uom being changed for some reason + let's keep it consistent with what will happen in the view since it is already using `rounding_method='HALF-UP'` (i.e. try your test case with 1.005 to see that it will be rounded to 1.01 in the view)

```suggestion
                rounding = move.product_id.uom_id.rounding
                diff = move.product_uom._compute_quantity(vals['qty_done'] - move_line.qty_done, move.product_id.uom_id, rounding_method='HALF-UP')
```

Reply:
Sorry this change doesn't work due to the _compute_quantity using precision_rounding instead of precision_digits :thinking: 

Try your original fix partially fixes your problem, but you will still have a problem with the diff if you do something like:

- original done_qty = 1.01
- updated done_qty = 1
- final svl of updated done_qty has moved quantity = .02

Either way, I believe it is up to the user to not use greater precision than their UoM is set to have  (i.e. they can add higher precision to avoid this bug) so this is a bug that we may not fix

Reply:
@ticodoo If I were a user, I would feel terrible about this because the margin of error is quite significant.
What do you think about combining my fix with `rounding_method='HALF-UP'` in `_compute_quantity` ?

Reply:
@ticodoo Please respond to my comment. Thanks!

Reply:
@lathuat1997 ok for me. Add the `rounding_method='HALF-UP'` to your fix  and also move `float_compare` before `float_is_zero` so that it follows our alphabetical imports standard: https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#imports

Note that your fix will not work in case the UoM in the move line is somehow force changed (i.e. through customizations), but it looks like this updating the done qty after picking is done feature is already not working correctly for the svls if the UoM of the move and move line don't match.

Reply:
@ticodoo Thank you very much!
I have adjusted according to your comment.

The case where the UoM of the `move` and `move line` don't match rarely occurs, and in that case, I think the value of `diff` is also not accurate. I think we can handle this case later.

I hope that you will merge this PR first. Thank you!